extension/refreshable_file/file_datasource.go

```diff
@@ -0,0 +1,79 @@
+package refreshable_file
+
+import (
+	"github.com/alibaba/sentinel-golang/extension"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+var (
+	logger = logging.GetDefaultLogger()
+)
+
+const refreshIntervalInSecond = 3
+
+type FileDataSource struct {
+	// the path of rule
+	// each property type corresponds to one file.
+	ruleFilePath string
+	handlers     []extension.DataSourceHandler
+}
+
+func FileDataSourceStarter(ruleFilePath string, handlers ...extension.DataSourceHandler) *FileDataSource {
+	ds := &FileDataSource{
+		ruleFilePath: ruleFilePath,
+		handlers:     handlers,
+	}
+	ds.Initialize()
+	go util.RunWithRecover(ds.listen, logger)
+	return ds
+}
+
+func (ds *FileDataSource) listen() {
```

Comment:
As you add two function for type DataSource interface, all the specific datasource should implement it.